src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkConfigurationHelpers.cs

```diff
@@ -19,26 +25,42 @@ public static IWebHostBuilder UseBenchmarksConfiguration(this IWebHostBuilder bu
             // Handle the transport type
             var webHost = builder.GetSetting("KestrelTransport");
 
-            if (string.Equals(webHost, "Sockets", StringComparison.OrdinalIgnoreCase))
+            Console.WriteLine($"KestrelTransport={webHost}");
+
+            if (string.Equals(webHost, "SocketPipe", StringComparison.OrdinalIgnoreCase))
             {
-                builder.UseSockets(options =>
+                builder.ConfigureServices(services =>
                 {
-                    if (int.TryParse(builder.GetSetting("threadCount"), out int threadCount))
-                    {
-                       options.IOQueueCount = threadCount;
-                    }
+                    services.AddSingleton<IConnectionListenerFactory, SocketPipeTransportFactory>();
```

Comment:
This should be the only thing you need to add, the other 2 types aren't required.